### ISSUE TITLE
surrounding lines region with empty lines

### ISSUE THREAD

**Issue by blaenk** [timcharper/evil-surround#67]
I would absolutely love to be able to do something like `V-S-RET`, that is, select some lines, press the surround operator and then give it an argument of RET so that it surrounds the lines with empty lines. I find myself doing this sometimes to space logically-grouped chunks of code. Doing this via the surround bindings feels the most natural way to do this, so I'm wondering if there's a way.

I tried doing it out of the box but it instead added little `^M` markers, which I guess is essentially two lines, or something.

Before:

``` elisp
  (use-package evil-surround
    :config ;; point is here
    (global-evil-surround-mode 1))
```

After:

``` elisp
  (use-package evil-surround
    ^M
    :config
    ^M
    (global-evil-surround-mode 1))
```

Expected:

``` elisp
  (use-package evil-surround

    :config

    (global-evil-surround-mode 1))
```

So next I tried to add my own surround pair (btw `^M` is the effect of `C-v RET`):

``` elisp
(push '(?^M . ("\n" . "\n")) evil-surround-pairs-alist)
```

But then I end up getting two lines instead of one, and the second one that appears on the bottom is indented, and the previously-following non-whitespace line is unindented.

Before:

``` elisp
  (use-package evil-surround
    :config ;; point is here
    (push '(?^M . ("\n" . "\n")) evil-surround-pairs-alist)
    (global-evil-surround-mode 1))
```

After `V-S-RET`:

``` elisp
  (use-package evil-surround


    :config

    ;; intended to here; the other empty lines are properly unindented. notice next is incorrectly unindented
(push '(?^M . ("\n" . "\n")) evil-surround-pairs-alist)
    (global-evil-surround-mode 1))
```

Perhaps this is outside of the scope of evil-surround. I would love it if I could use these bindings though. If this isn't possible via evil-surround, could I perhaps hook on my own function that does this to some evil-surround map so that I can trigger it when I press `S` and `RET`?

**Comment by blaenk** [comment]
Any thoughts? I keep wishing I could do this :smile:

**Comment by timcharper** [comment]
Try using an empty string for your surround pairs?

**Comment by blaenk** [comment]
That takes care of the problem of it "inserting" two lines on each side. The problems that remain are:

1. the empty line on the bottom of the surrounded text isn't actually empty; the original indentation of the line that got "shifted" down remains
2. that line that got "shifted" down is now unindented

Before (intention is to surround the `evil-surround-pairs-alist` line with empty lines):

``` elisp
(use-package evil-surround
  :config
  (setq-default
   evil-surround-pairs-alist
   (cons '(?
 . ("" . "")) evil-surround-pairs-alist))
  (global-evil-surround-mode 1))
```

After:

``` elisp
(use-package evil-surround
  :config
  (setq-default

   evil-surround-pairs-alist
   ;; whitespace up to this point
(cons '(?
 . ("" . "")) evil-surround-pairs-alist))
  (global-evil-surround-mode 1))
```

In other words, the result is as if the point were placed in between the indentation and the first non-whitespace character of the line under the surrounded text, i.e. where the `|` appears here right before the `(cons`:

``` elisp
(use-package evil-surround
  :config
  (setq-default
   evil-surround-pairs-alist
   |(cons '(?
 . ("" . "")) evil-surround-pairs-alist))
  (global-evil-surround-mode 1))
```

Then it's as if one were to press `RET` so that `(cons` was now on the next line, _unindented_.

Perhaps this ability requires special casing?

I would take a shot at implementing it myself as its own separate function, but I'd love to be able to do it on the evil-surround map, so that I can do e.g. `V-S-RET`, i.e. bind `RET` on the evil-surround map. Would that be possible? Then I could take it from there (getting the selected region, etc.).

**Comment by blaenk** [comment]
I would take a shot at implementing it myself as its own separate function, but I'd love to be able to do it on the evil-surround map, so that I can do e.g. `V-S-RET`, i.e. bind `RET` on the evil-surround map. Would that be possible? Then I could take it from there (getting the selected region, etc.).

**Comment by blaenk** [comment]
Yeah it seems to me like it's due to [indentation logic](https://github.com/timcharper/evil-surround/blob/master/evil-surround.el#L297-L313), which is the correct thing to do in most cases, but perhaps not in this case.

**Comment by blaenk** [comment]
Fixed by #73
